src/framework/components/sound/component.js

```diff
@@ -414,6 +414,86 @@ class SoundComponent extends Component {
         return this._slots[name];
     }
 
+    /**
+     * Returns true if the slot with the specified name is currently playing.
+     *
+     * @param {string} name - The name of the {@link SoundSlot} to look for.
+     * @returns {boolean} True if the slot with the specified name exists and is currently playing.
+     */
+    isPlaying(name) {
+        if (!this.enabled || !this.entity.enabled) {
+            return false;
+        }
+
+        const slot = this._slots[name];
+        if (!slot) {
+            Debug.warn(`Trying to get information from sound slot with name ${name} which does not exist`);
+            return false;
+        }
+
+        return slot.isPlaying;
```

Comment:
Quite a lot of code duplication here. Could you have something like:

```
    _getSlotProperty(slotName, property) {
        if (!this.enabled || !this.entity.enabled) {
            return false;
        }

        const slot = this._slots[slotName];
        if (!slot) {
            Debug.warn(`Trying to get information from sound slot with name ${name} which does not exist`);
            return false;
        }

        return slot[property];
    }
```

Reply:
It's done!